src/app/components/login/login.component.spec.ts

```
@@ -0,0 +1,279 @@
+    authService = TestBed.inject(AuthService) as jasmine.SpyObj<AuthService>;
+    router = TestBed.inject(Router) as jasmine.SpyObj<Router>;
+    snackBar = TestBed.inject(MatSnackBar) as jasmine.SpyObj<MatSnackBar>;
+  });
+
+  beforeEach(() => {
+    // Reset spies before each test
+    mockAuthService.login.calls.reset();
+    mockAuthService.clearError.calls.reset();
+    mockRouter.navigate.calls.reset();
+    mockSnackBar.open.calls.reset();
+  });
```

Biome (1.9.4)

[error] Disallow duplicate setup and teardown hooks.

Disallow beforeEach duplicacy inside the describe function.

(lint/suspicious/noDuplicateTestHooks)